Update CHANGELOG and up version to the 1.0.0 (#73)
heremaps/here-olp-sdk-typescript: CHANGELOG.md

* **Breaking Change** The `getTile` method was removed from `CatalogClient`. Now, to get layer data, use `VersionedLayerClient.getData()` or `VolatileLayerClient.getData()`.
* **Breaking Change** The `getAgregatedTile` method was removed from `CatalogClient`. Now, to get partitions metadata and layer data, use `VersionedLayerClient.getPartitions()` and `VersionedLayerClient.getData()`
or `VolatileLayerClient.getPartitions()` and `VolatileLayerClient.getData()`.
* **Breaking Change** The `getSchema` and `getSchemaDetails` methods were removed from `CatalogClient`. Now, to get schemas and schema metadata, use the `ArtifactClient` class.
* **Breaking Change** The `HypeDataProvider` class was removed. Now, catalog and layer clients can be initialized without additional classes.
* **Breaking Change** The `DatastoreClient` class was removed. Now, `CatalogClient` can be initialized without additional classes.

**olp-sdk-dataservice-api**

* Fixed issues with requests to the Blob REST APIs.

**olp-sdk-authentication**

> * Added posibillity to read credentials from file when using the NodeJs.

Reviewer: ```suggestion
* Added a posibillity to read credentials from a file when using Node.js.
```

## v1.0.0 (02/12/2019)

**olp-sdk-dataservice-read**

> * Added new class `OlpClientSetting` with `KeyValueCache` instance to be used for context

Author: updated

> * Added new class `VersionedLayerClient` that is used to access versioned layers in OLP. Class implements `GetData`, `GetPartitions` methods.

Author: updated

> * Added new class `VolatileLayerClient` that is used to access volatile layers in OLP. Class implements `GetPartitions`, `GetData` methods.

Author: updated

> * Added new class `ArtifactClient` that is used to access artifact service in OLP. Class implements `getSchemaDetails`, `getSchema` methods.

Reviewer: ```suggestion
* Added the `ArtifactClient` class that is used to access the artifact service on OLP. This class implements the `getSchemaDetails` and `getSchema` methods.
```

Author: updated

* Added new class `CatalogClient` that is used to access catalog metada in OLP. Class implements `getCatalog`, `getLatestVersion`, `getVersions` methods.

> * Added new class `ConfigClient` that is used to access config service in OLP. Class implements `getCatalogs` method.

Reviewer: ```suggestion
* Added the `ConfigClient` class that is used to access the configuration service on OLP. This class implements the `getCatalogs` method.
```

**Breaking Change**

> * CatalogClient constructor changed. `OlpClientSettings` must be passed by value, `KeyValueCache` is now part of `OlpClientSettings`.

Reviewer: ```suggestion
* The `CatalogClient` constructor changed. `OlpClientSettings` must be passed by a value, and `KeyValueCache` is now a part of `OlpClientSettings`.
```

## v1.0.0 (02/12/2019)

**Common**

* Project structure was changed, public API aligned to the HERE OLP SDK for C++.

> * Unit tests was improved

Reviewer 2: Not sure we should write this. I would not write this.

Author: Removed

**olp-sdk-dataservice-read**

> * Added new class `OlpClientSetting` with `KeyValueCache` instance to be used for context

Reviewer 2: ```suggestion
* Added new class `OlpClientSetting` with `KeyValueCache` instance to be used for context.
```

Author: updated

> * Added new class `VersionedLayerClient` that is used to access versioned layers in OLP. Class implements `GetData`, `GetPartitions` methods.

Reviewer 2: ```suggestion
* Added new class `VersionedLayerClient` that is used to access versioned layers in OLP. Class implements `getData` and `getPartitions` methods.
```

Author: updated

> * Added new class `VolatileLayerClient` that is used to access volatile layers in OLP. Class implements `GetPartitions`, `GetData` methods.

Reviewer 2: ```suggestion
* Added new class `VolatileLayerClient` that is used to access volatile layers in OLP. Class implements `getPartitions`, `getData` methods.
```

Author: updated

* Added new class `ArtifactClient` that is used to access artifact service in OLP. Class implements `getSchemaDetails`, `getSchema` methods.

> * Added new class `CatalogClient` that is used to access catalog metada in OLP. Class implements `getCatalog`, `getLatestVersion`, `getVersions` methods.

Reviewer 2: `CatalogClient` was there before you just modified it, right? Please write what dropped and what changed.

Reviewer: ```suggestion
* Added the `CatalogClient` class that is used to access catalog metada on OLP. This class implements the `getCatalog`, `getLatestVersion`, and `getVersions` methods.
```

Author: updated

* Added new class `ConfigClient` that is used to access config service in OLP. Class implements `getCatalogs` method.

**Breaking Change**

* CatalogClient constructor changed. `OlpClientSettings` must be passed by value, `KeyValueCache` is now part of `OlpClientSettings`.

> * HypeDataProvider removed.

Reviewer 2: Please elaborate: which methods were moved where, which were removed completely and why.

Author: updated

> * DatastoreClient removed.

Reviewer 2: Same here.

Author: updated

> * Methods getTile and getAgregatedTile removed.

Reviewer 2: From were removed and why? What is the alternative method to use?

Author: updated

> 

Reviewer 2: Please always keep the same structure:

**Common**
**olp-sdk-dataservice-api**
**olp-sdk-authentication**
**olp-sdk-dataservice-read**
**olp-sdk-dataservice-write**
**olp-sdk-fetch**

Author: updated

## v1.0.0 (02/12/2019)

**olp-sdk-dataservice-read**

* Added the `OlpClientSetting` class with the `KeyValueCache` instance.
* Added the `VersionedLayerClient` class that is used to access versioned layers on OLP. This class implements the `getData` and `getPartitions` methods.
* Added the `VolatileLayerClient` class that is used to access volatile layers on OLP. This class implements the `getPartitions` and `getData` methods.
* Added the `ArtifactClient` class that is used to access the artifact service on OLP. This class implements the `getSchemaDetails` and `getSchema` methods.
* Added the `ConfigClient` class that is used to access the configuration service on OLP. This class implements the `getCatalogs` method.
* Added the `QueryClient` class that is used to access the query service on OLP. This class implements the `fetchQuadTreeIndex` method.
* Added a possibility to abort requests using `AbortSignal`.

> * Added a possibility to set a billing tag in each request.

Reviewer 2: ```suggestion
* Added a possibility to set an optional billing tag in each request.
```

* Added `QuadKeyUtils` that has a set of utilities and functions intended for work with quadkeys: convert a quadkey to a Morton code, convert a Morton code to a quadkey, and validate a quadkey.
* Added the `RequestFactory` class that is used for look-up requests. This class implements the static `create` method that returns the built `DatastoreDownloadManager` instance with a correct base URL to service, and the `getBaseUrl` method that caches and returns base URLs to services.

> **Breaking Changes**

Reviewer 2: **olp-sdk-dataservice-read** is a module while **Breaking Changes** is not. Not sure if it really denotes that the breaking changes actually belong to olp-sdk-dataservice-read module section. I would favour something more like the one from v0.9.0 and CPP SDK:

* **Breaking Change** The `CatalogClient` constructor changed. Now, pass `OlpClientSettings` by a value. Also, `KeyValueCache` is now a part of `OlpClientSettings`.

NOTE the missing `-`

> * The `CatalogClient` constructor changed. Now, pass `OlpClientSettings` by a value. Also, `KeyValueCache` is now a part of `OlpClientSettings`.

Reviewer 2: ```suggestion
* The `CatalogClient` constructor changed. Users must pass now `OlpClientSettings` by value not by reference anymore. Removed `KeyValueCache` as it is now a part of `OlpClientSettings`.
```

* The `getLayer` method was removed from `CatalogClient`. Now, to initialize a deprecated `CatalogLayer`, use the `VersionedLayerClient` or `VolatileLayerClient` classes.
* The `getTile` method was removed from `CatalogClient`. Now, to get layer data, use `VersionedLayerClient.getData()` or `VolatileLayerClient.getData()`.
* The `getAgregatedTile` method was removed from `CatalogClient`. Now, to get partitions metadata and layer data, use `VersionedLayerClient.getPartitions()` and `VersionedLayerClient.getData()`
or `VolatileLayerClient.getPartitions()` and `VolatileLayerClient.getData()`.
* The `getSchema` and `getSchemaDetails` methods were removed from `CatalogClient`. Now, to get schemas and schema metadata, use the `ArtifactClient` class.
* The `HypeDataProvider` class was removed. Now, catalog and layer clients can be initialized without additional classes.
* The `DatastoreClient` class was removed. Now, `CatalogClient` can be initialized without additional classes.

**olp-sdk-dataservice-api**

> * Fixed issues with requests to the blob storages

Reviewer 2: ```suggestion
* Fixed issues with requests to the blob storages.
```
What is blob storages? Do you mean Blob REST API?

Author: yes

**olp-sdk-authentication**

> * Added posibillity to read credentials from file on the NodeJs.

Reviewer 2: ```suggestion
* Added posibillity to read credentials from file when using NodeJs.
```

> * Added posibillity to read credentials from file on the NodeJs.

Reviewer: ```suggestion
* Added a possibillity to read credentials from a file when using Node.js.
```

## v0.9.2-beta (28/10/2019)

> * Changed prepare process before publishing

Reviewer 2: Please adapt below to look similar to above or vice versa.
Add dots at the end of the line, remove extra newlines.
Also v0.9.1-beta uses breaking change differently then above. Which one should be the case now?
